src/pretty.js

```diff
@@ -104,10 +104,12 @@ function _renderToStringPretty(
 	// components
 	if (typeof nodeName === 'function') {
 		isComponent = true;
-		if (opts.shallow && (inner || opts.renderRootComponent === false)) {
-			nodeName = getComponentName(nodeName);
-		} else if (nodeName === Fragment) {
-			const children = [];
+
+		let isFragment = nodeName === Fragment,
+			isShallow = opts.shallow && (inner || opts.renderRootComponent === false);
+
+		if (isFragment || (isShallow && isFragment)) {
+			let children = [];
```

Comment:
nit: let's keep it as `const`, reduces the diff

```suggestion
			const children = [];
```